harness/determined/pytorch/_reducer.py

```diff
@@ -1,9 +1,176 @@
+                # returned by the final self.accumulate() on each slot
+                sums, counts = zip(*per_slot_metrics)
+                return sum(sums) / sum(counts)
+
+        class MyPytorchTrial(pytorch.PyTorchTrial):
+            ...
+            def evaluate_batch(...):
+                ...
+                return {"my_averageable_metric": metric_tensor, ...}
+
+            def evaluation_reducer(...):
+                return {"my_averageable_metric": MyAvgMetricReducer, ...}
+
+    See also: :meth:`determined.pytorch.PyTorchTrial.evaluation_reducer`.
+    """
+
+    @abc.abstractmethod
+    def accumulate(self, metric: Any) -> Any:
+        """
+        accumulate is called for each batch in the evaluation dataset.  Batches will be distributed
+        across slots, so accumulate will be called many times on each slot.
+
+        accumulate should return the accumulated state.  After evaluation is complete, the final
+        return value of accumulate will become an element of the per_slot_metrics argument to
+        cross_slot_reduce.
+
+        In the example of the calculating a distributed mean, accumulate might keep a running sum
+        and a count of metrics received:
+
+        .. code:: python
+
+            def accumulate(self, metric):
+                self.sum += metric
+                self.count += 1
+                return self.sum, self.count
+        """
+        pass
+
+    @abc.abstractmethod
+    def cross_slot_reduce(self, per_slot_metrics: List[Any]) -> Any:
+        """
+        cross_slot_reduce is called on the list of results from the final call to accumulate on
+        each slot.  per_slot_metrics will be a list of length N, where N is the number of slots in
+        the trial (or 1 in non-distributed training).  cross_slot_reduce must return the final
+        metric.
+
+        In the example of calculating a distributed mean, cross_slot_reduce might recieve a list of
+        (sum, count) tuples and it would calculate the overall mean.
+
+        .. code:: python
+
+            def cross_slot_reduce(self, per_slot_metrics):
+                sums, counts = zip(*per_slot_metrics)
+                return np.array(sum(sums) / sum(counts))
+        """
+        pass
+
+
+class _SimpleMetricReducer(MetricReducer):
+    """_SimpleMetricReducer takes a one-step reducer function and converts it to a MetricReducer."""
+
+    def __init__(self, reduce_fn: Callable[[List[Any]], Any]):
+        self.updates = []  # type: List[Any]
+        self.reduce_fn = reduce_fn
+
+    def accumulate(self, metric: Any) -> List[Any]:
+        self.updates.append(metric)
+        return self.updates
+
+    def cross_slot_reduce(self, per_slot_metrics: List[List[Any]]) -> Any:
+        flat_metrics = [item for sublist in per_slot_metrics for item in sublist]
+        return self.reduce_fn(flat_metrics)
+
+
+class AvgMetricReducer(MetricReducer):
+    def __init__(self) -> None:
+        self.sum = None
```

Comment:
blocking: I think it's safe to initialize the sum to be 0.

Reply:
happy to address this after the following checkmark from the PR description is addressed:

> Ensure the behaviors of the Avg/Sum/Max/Min MetricReducers match the old Reducer.AVG/SUM/MAX/MIN behaviors

Because starting with `sum=0` is assuming a certain shape.